libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.spec.tsx

```diff
@@ -0,0 +1,83 @@
+import { act, findByTestId, render, waitFor } from '@testing-library/react'
+import { wrapWithReactHookForm } from '__tests__/utils/wrap-with-react-hook-form'
+import { BuildModeEnum, BuildPackLanguageEnum } from 'qovery-typescript-axios'
+import { upperCaseFirstLetter } from '@console/shared/utils'
+import PageSettingsGeneral, { PageSettingsGeneralProps } from './page-settings-general'
+
+describe('PageSettingsGeneral', () => {
+  const props: PageSettingsGeneralProps = {
+    watchBuildMode: BuildModeEnum.DOCKER,
+    onSubmit: jest.fn((e) => e.preventDefault()),
+  }
+
+  const defaultValues = (mode = BuildModeEnum.DOCKER) => ({
+    name: 'hello-world',
+    build_mode: mode,
+    buildpack_language: BuildPackLanguageEnum.CLOJURE,
+    dockerfile_path: 'Dockerfile',
+  })
+
+  it('should render successfully', async () => {
+    const { baseElement } = render(wrapWithReactHookForm(<PageSettingsGeneral {...props} />))
+    expect(baseElement).toBeTruthy()
+  })
+
+  it('should render the form with docker section', async () => {
+    const { baseElement } = render(
+      wrapWithReactHookForm(<PageSettingsGeneral {...props} />, {
+        defaultValues: defaultValues(),
+      })
+    )
+
+    let name: any, buildMode: any, dockerfile: any
+    await act(async () => {
+      name = await findByTestId(baseElement, 'input-name')
+      buildMode = await findByTestId(baseElement, 'input-select-mode')
+      dockerfile = await findByTestId(baseElement, 'input-text-dockerfile')
+    })
+    expect(name.getAttribute('value')).toBe('hello-world')
```

Comment:
Instead of find or get + getAttribute value, you could do a :
`getByDisplayedValue`
This method looks for an input with a specific value and fail the test if it does not find it.